src/Microsoft.AspNet.Mvc.Localization/ViewLocalizer.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Globalization;
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.Framework.Internal;
+using Microsoft.Framework.Localization;
+using Microsoft.Framework.Runtime;
+
+namespace Microsoft.AspNet.Mvc.Localization
+{
+    /// <summary>
+    /// A <see cref="HtmlLocalizer"/> that provides localized strings for views.
+    /// </summary>
+    public class ViewLocalizer : IViewLocalizer, ICanHasViewContext
+    {
+        private readonly IHtmlLocalizerFactory _localizerFactory;
+        private readonly string _applicationName;
+        private IHtmlLocalizer _localizer;
+
+        /// <summary>
+        /// Creates a new <see cref="ViewLocalizer"/>.
+        /// </summary>
+        /// <param name="localizerFactory">The <see cref="IHtmlLocalizerFactory"/>.</param>
+        /// <param name="applicationEnvironment">The <see cref="IApplicationEnvironment"/>.</param>
+        public ViewLocalizer(
+            [NotNull] IHtmlLocalizerFactory localizerFactory,
+            [NotNull] IApplicationEnvironment applicationEnvironment)
+        {
+            _applicationName = applicationEnvironment.ApplicationName;
+            _localizerFactory = localizerFactory;
+        }
+
+        /// <inheritdoc />
+        public LocalizedString this[string name] => _localizer[name];
```

Comment:
My concern here was slightly different though that `[NotNull]` is a good catch.

`_localizer` remains `null` after the constructor has completed its work. Unless `Contextualize()` is called, the instance's initialization remains incomplete. In the same scenario `HtmlHelper`'s properties will throw an `InvalidOperationException` with the `Message` "Must call 'Contextualize' method before using this HtmlHelper instance.".